cargo-marker/src/main.rs

```diff
@@ -84,14 +76,15 @@ fn run_check(args: &clap::ArgMatches, config: Option<Config>, flags: &Flags) ->
     }
 
     // If this is a dev build, we want to rebuild the driver before checking
-    if flags.dev_build {
-        backend::driver::install_driver(false, flags.dev_build, "")?;
-    }
+    #[cfg(debug_assertions)]
+    backend::driver::install_driver(false, "")?;
 
     // Configure backend
-    let toolchain = backend::toolchain::Toolchain::try_find_toolchain(flags.dev_build, flags.verbose)?;
+    // FIXME(xFrednet): Implement better logging and remove verbose boolean in
+    // favor of debug logging.
+    let toolchain = backend::toolchain::Toolchain::try_find_toolchain(false)?;
     let backend_conf = backend::Config {
-        dev_build: flags.dev_build,
+        dev_build: cfg!(debug_assertions),
```

Comment:
IIRC there are several places where dev build flag influences the discovery of the toolchain and driver.

I'd prefer if there was a magic env variable e.g. `MARKER_DEV_MODE=1` that could be used for conditions when a dev mode is used. This wouldn't require building in release or changing code temporarily to [de]activate the dev mode.

Reply:
Part of this change was to have a unified way across the entire project to enter dev mode. Having an environment value could work, but would be a bit weird, as it would always need to be set when interacting with the project. Or am I missing another option I'm missing?

Reply:
I guess you are right. Maybe MARKER_RELEASE_MODE could then be used as an override to opt out.

Reply:
I feel like it's too easy for a user to accidentally install it without setting the environment value.